addons/web_editor/static/src/components/media_dialog/image_selector.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<templates id="template" xml:space="preserve">
+<t t-name="web_editor.AutoResizeImage" owl="1">
+    <div t-ref="auto-resize-image-container" class="o_existing_attachment_cell o_we_image position-relative align-items-center justify-content-center" t-att-class="{ o_we_attachment_optimized: props.isOptimized, o_loaded: state.loaded, o_we_attachment_selected: props.selected }" t-on-click="props.onImageClick">
+        <RemoveButton t-if="props.isRemovable" model="props.model" remove="() => this.remove()"/>
+        <div class="o_we_media_dialog_img_wrapper">
+            <img t-ref="auto-resize-image" class="img img-fluid o_we_attachment_highlight" t-att-src="props.src" t-att-alt="props.altDescription" t-att-title="props.title" loading="lazy"/>
+            <a t-if="props.author" class="o_we_media_author" t-att-href="props.authorLink" target="_blank" t-esc="props.author"/>
+        </div>
+        <span t-if="props.isOptimized" class="badge badge-success">Optimized</span>
+    </div>
+</t>
+
+<t t-name="web_editor.ImagesListTemplate" owl="1">
+    <div class="o_we_existing_attachments o_we_images d-flex flex-wrap w-100 justify-content-between align-items-stretch my-0">
+        <t t-if="!hasContent and !isFetching">
+            <div t-if="state.needle" class="o_nocontent_help">
+                <p class="o_empty_folder_image">No images found.</p>
+                <p class="o_empty_folder_subtitle">You can upload images with the button located in the top left of the screen.</p>
+            </div>
+            <div t-else="" class="o_we_search_prompt">
+                <h2>Get the perfect image by searching in our library of copyright free photos and illustrations.</h2>
+            </div>
+        </t>
+        <t t-else="">
+            <t t-if="['all', 'database'].includes(state.searchService)" t-foreach="state.attachments" t-as="attachment" t-key="attachment.id">
```

Comment:
If you put `t-if` and `t-foreach` on the same item, `t-if` is called for each loop step (you can use `attachment` in the condition here). So here, it's wrong, that's not what you want (worth checking all your foreach? :grimacing:). And even in the good case (where you want the `if` for each loop step), I'd say it's a bad practice, better put the `t-if` on its own element.